src/upload.ts

```diff
@@ -437,25 +465,36 @@ export function splitSizeIntoChunkAlignedParts(
   partCount: number,
   chunkSize: number
 ): Array<{ start: number; end: number }> {
+  if (partCount < 1) {
+    throwValidationError("partCount", partCount, "option", "greater than or equal to 1");
+  }
+  if (chunkSize < 1) {
+    throwValidationError("chunkSize", chunkSize, "option", "greater than or equal to 1");
+  }
+
   const partSizes = new Array(partCount).fill(0);
-  // The leftover size that must go into the last part.
-  const leftover = totalSize % chunkSize;
 
   // Assign chunks to parts in order, looping back to the beginning if we get to
   // the end of the parts array.
-  let lastPart = -1;
-  for (let i = 0; i < Math.floor(totalSize / chunkSize); i++) {
+  const numFullChunks = Math.floor(totalSize / chunkSize);
+  for (let i = 0; i < numFullChunks; i++) {
     partSizes[i % partCount] += chunkSize;
-    if (i > lastPart) lastPart = i;
   }
 
-  if (lastPart === -1) {
-    // No parts were visited, so assign to the last part.
-    partSizes[partCount - 1] += leftover;
-  } else {
-    // Assign the leftover to the part after the last part that was visited, or
-    // the last part in the array if all parts were used.
-    partSizes[Math.min(lastPart + 1, partCount - 1)] += leftover;
+  // The leftover size that must go into the last part.
+  const leftover = totalSize % chunkSize;
+  // If there is non-chunk-aligned leftover, add it.
+  if (leftover > 0) {
+    let lastIndex;
+    if (numFullChunks === 0) {
+      // No parts were visited, so just assign to the last part.
+      lastIndex = partCount - 1;
```

Comment:
Q: If no parts were visited, that means that the total size of the upload is under the chunk size. That would mean that all parts, except the last one, will be empty. That's perfectly fine by me, as long as TUS won't complain about it. Is this the case, both in terms of part sizes and TUS behaviour?

Reply:
Honestly I've never run into this because we disable parallel uploads when we're under a chunk. I don't know how it behaves, but this should just throw as it's an unexpected flow path.

Reply:
PS having e.g. two parts and only using one of them is fine, this has been tested. I'll document that in a comment.